crates/red_knot_python_semantic/resources/mdtest/sys_version_info.md

```diff
@@ -0,0 +1,137 @@
+# `sys.version_info`
+
+## The type of `sys.version_info`
+
+The type of `sys.version_info` is `sys._version_info`, at least according to typeshed's stubs (which
+we treat as the single source of truth for the standard library). This is quite a complicated type
+in typeshed, so there are many things we don't fully understand about the type yet; this is the
+source of several TODOs in this test file. Many of these TODOs should be naturally fixed as we
+implement more type-system features in the future.
+
+```py
+import sys
+
+reveal_type(sys.version_info)  # revealed: _version_info
+```
+
+## Literal types from comparisons
+
+Comparing `sys.version_info` with a 2-element tuple of literal integers always produces a `Literal`
+type:
+
+```py
+import sys
+
+reveal_type(sys.version_info >= (3, 8))  # revealed: Literal[True]
+reveal_type((3, 8) <= sys.version_info)  # revealed: Literal[True]
+
+reveal_type(sys.version_info > (3, 8))  # revealed: Literal[True]
+reveal_type((3, 8) < sys.version_info)  # revealed: Literal[True]
+
+reveal_type(sys.version_info < (3, 8))  # revealed: Literal[False]
+reveal_type((3, 8) > sys.version_info)  # revealed: Literal[False]
+
+reveal_type(sys.version_info <= (3, 8))  # revealed: Literal[False]
+reveal_type((3, 8) >= sys.version_info)  # revealed: Literal[False]
+
+reveal_type(sys.version_info == (3, 8))  # revealed: Literal[False]
+reveal_type((3, 8) == sys.version_info)  # revealed: Literal[False]
+
+reveal_type(sys.version_info != (3, 8))  # revealed: Literal[True]
+reveal_type((3, 8) != sys.version_info)  # revealed: Literal[True]
+```
+
+## Non-literal types from comparisons
+
+Comparing `sys.version_info` with tuples of other lengths will sometimes produce `Literal` types,
+sometimes not:
+
+```py
+import sys
+
+reveal_type(sys.version_info >= (3, 8, 1))  # revealed: bool
+reveal_type(sys.version_info >= (3, 8, 1, "final", 0))  # revealed: bool
+
+# TODO: this is an invalid comparison (`sys.version_info` is a tuple of length 5)
+# Should we issue a diagnostic here?
```

Comment:
I don't think it's invalid. It's not a runtime error to compare tuples of different lengths, so I don't think it should be a type diagnostic, either. Such a comparison can be either true or false.

Reply:
"Invalid" is probably too strong a word, but it seems pretty clear that the user has made a mistake here, right? It probably shouldn't be a type-check diagnostic, but a "type aware lint diagnostic" might be helpful for the user.

Does that make sense? Shall I reword the TODO comment along those lines?

Reply:
Yeah, I can see a case for a special-cased warning diagnostic here, but it seems a bit subtle. The existence of additional elements can actually change the result of the comparison! So while it seems like probably a mistake, it's neither a no-op, nor does it render the comparison always-true or always-false. So it's a bit tricky to figure out how to word a "this is wrong" message.

I don't think it's critical to update the comment, although I do think "invalid" is too strong. I don't think a diagnostic for this is a priority, but I don't mind a comment suggesting that we could consider it at some point.

Reply:
> The existence of additional elements can actually change the result of the comparison!

I'm not even suggesting that we implement a generalised rule for tuples, to be clear! (Though we could, of course.) `sys.version_info` is an instance of a singleton tuple subclass, so we can implement a rule specifically for "comparing a tuple of >5 elements against instances of `sys._version_info`" if we want to. Comparisons against `sys.version_info` are common enough that this could be valuable. (Ruff already has one or two rules that aim to catch common errors in `sys.version_info` comparisons.)